### PR TITLE
[CI] Add new CI stage for testing cugraph

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

https://github.com/dmlc/dgl/pull/4166 depends on this for CI test.

As suggested, the base is `nvcr.io/nvidia/pytorch:22.04-py3` which includes `cuda11.6`, `pytorch1.12.0a0+bd13bc6`, `rapids 22.02`.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -426,6 +448,30 @@ pipeline {
...
+                  args "--runtime nvidia --shm-size=8gb"
+                  alwaysPull false
+                }
+              }
+              stages {
+                stage('Torch GPU Unit test') {
+                  steps {
+                    sh 'nvidia-smi'
+                    sanity_check_for_nvidia_image()
+                    unit_test_linux('pytorch', 'gpu_nv')
```
Comment (jermainewang) [inline Jenkinsfile:465]: We only need to test cugraph unittests. Perhaps also name the stage as `Torch+cuGraph`,

File: tests/scripts/task_unit_test.sh [hunk 2]
```diff
@@ -16,21 +16,27 @@ if [ $# -ne 2 ]; then
     fail "Error: must specify backend and device"
 fi
 
+TEST_DEVICE=$2
```
Comment (jermainewang) [inline tests/scripts/task_unit_test.sh:19]: There is no need to modify the old test scripts. Just add a new script and test cugraph related UT.

Comment (Rhett-Ying) [reply tests/scripts/task_unit_test.sh:19]: I added a new directory to test cugraph related UT only, though I utilize existing `task_unit_test.sh`. An extra `test.sh` is very like to existing one. What do you think?

Comment (jermainewang) [reply tests/scripts/task_unit_test.sh:19]: Having a lot of if/else in `task_unit_test.sh` makes it hard to maintain in the future. Please add a new script.

File: tests/scripts/build_dgl.sh [hunk 3]
```diff
@@ -38,7 +40,13 @@ popd
 pushd python
 for backend in pytorch mxnet tensorflow
 do
-conda activate "${backend}-ci"
+if [[ $1 == "cugraph" ]]; then
```
Comment (jermainewang) [inline tests/scripts/build_dgl.sh:43]: refactor suggestion:
```
if [[ $1 == "cugraph" ]]; then
    ...
else
    for backend in pytorch mxnet tensorflow; do
        ...
    done
fi
```

File: tests/scripts/task_unit_test.sh (hunk 2 above)
Comment (Rhett-Ying) [reply tests/scripts/task_unit_test.sh]: done

File: tests/scripts/build_dgl.sh (hunk 3 above)
Comment (Rhett-Ying) [reply tests/scripts/build_dgl.sh:41]: done

**Review by jermainewang — APPROVED** [review]

**Pull request merged**